### PR TITLE
Archive

### PR DESCRIPTION
**Pull request by arpit01923**
Functionalities :
- add to archive
- unarchive
- delete from archive

Files to check :

- src/components/notecontainer/notecontainer.jsx
- src/pages/archive/archive.jsx
- src/reducer/notearray.jsx

### CODE REVIEW

**Review by deepak29-git — APPROVED** [review]
your folder structure and variable name is good

File: src/components/notecontainer/notecontainer.jsx [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+        `/api/notes/archives/${id}`,
+        { note },
+        { headers: { authorization: token } }
+      );
+      noteArrayDispatch({
+        type: "ADD_TO_ARCHIVE",
+        payload: response.data,
+      });
+    } catch (e) {
+      console.log(e);
```
Comment (hiteshverma27) [inline src/components/notecontainer/notecontainer.jsx:25]: remove all the logs before pushing, you can use toastify and give user an error message if something goes wrong

File: src/components/notecontainer/notecontainer.jsx [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+    try {
+      const response = await axios.delete(`/api/archives/delete/${id}`, {
+        headers: { authorization: token },
+      });
+      noteArrayDispatch({
+        type: "DELETE_FROM_ARCHIVE",
+        payload: response.data.archives,
+      });
+    } catch (e) {
+      console.log(e);
```
Comment (hiteshverma27) [inline src/components/notecontainer/notecontainer.jsx:39]: same here

File: src/pages/archive/archive.jsx [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
 export const Archive = () => {
+  const { noteArrayState } = useNote();
   return (
     <>
-      <Navbar />
       <section className="home">
         <Sidebar />
-        <NoteContainer />
+        <main className="content">
+          {noteArrayState.archives.map((archive, index) => (
```
Comment (hiteshverma27) [inline src/pages/archive/archive.jsx]: you can destructure archive so that you have to not write ```archive.``` everytime

**Review by hiteshverma27 — APPROVED** [review]
left some suggestions, rest looks good

File: src/components/notecontainer/notecontainer.jsx (hunk 1 above)
Comment (arpit01923) [reply src/components/notecontainer/notecontainer.jsx:25]: I will add toastify component at last

File: src/components/notecontainer/notecontainer.jsx (hunk 2 above)
Comment (arpit01923) [reply src/components/notecontainer/notecontainer.jsx:39]: add it at last

File: src/pages/archive/archive.jsx (hunk 3 above)
Comment (arpit01923) [reply src/pages/archive/archive.jsx]: done

**Pull request merged**